### PR TITLE
Add terraform module for ECS

### PR DESCRIPTION
**Pull request by rdettai**
### Description

Add Terraform scripts with:
- a module that can be used to deploy a general purpose Quickwit setup using ECS and RDS
- an self contained example stack that uses that module and includes supporting network configurations (VPC and bastion)

### How was this PR tested?

Executed on AWS with various configurations.

### CODE REVIEW

File: distribution/ecs/example/bastion.tf [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
```
Comment (trinity-1686a) [inline distribution/ecs/example/bastion.tf:44]: reading [this](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group#removing-all-ingress-and-egress-rules), it looks like `egress  = []` is the more natural way to write that

File: distribution/ecs/example/kafka.tf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      mount_points = [
+        {
+          sourceVolume  = "quickwit-keys"
+          containerPath = "/quickwit/keys"
+        }
+      ]
+      secrets = [
+        {
+          name      = "CA_PEM"
+          valueFrom = "arn:aws:secretsmanager:eu-west-1:542709600413:secret:your_kafka_ca_pem"
```
Comment (trinity-1686a) [inline distribution/ecs/example/kafka.tf:38]: (is that a well known arn or something?)

File: distribution/ecs/example/logging.tf [hunk 3]
```diff
@@ -0,0 +1,94 @@
+# Example configuration for pushing ECS logs to Datadog
```
Comment (trinity-1686a) [inline distribution/ecs/example/logging.tf:1]: sending logs to dd as qw strikes me as odd

**Review by trinity-1686a — COMMENTED** [review]
looks sensible, though I lack knowledge in terraform to confidently review that kind of PR

File: distribution/ecs/example/kafka.tf (hunk 2 above)
Comment (rdettai) [reply distribution/ecs/example/kafka.tf:38]: no, these are example values. Not sure how to make that more obvious

Comment (rdettai) [reply distribution/ecs/example/kafka.tf]: using 123456789 to make it more obvious that this is an example

File: distribution/ecs/example/bastion.tf (hunk 1 above)
Comment (rdettai) [reply distribution/ecs/example/bastion.tf:44]: my understanding of the docs is that `egress = []` means "no egress rule", i.e "no egress allowed". The rule configured here is the exact opposite, it means "all egress allowed".

Comment (rdettai) [reply distribution/ecs/example/bastion.tf:44]: image

**Review by trinity-1686a — APPROVED** [review]

**Pull request merged**